File: Commercial/Commercial_Controller/Commercial_Controller/Request.cs
using System;

namespace Commercial_Controller
{
    class Request
    {
        #region FIELDS
        private int _floor;
        private string _direction;
        #endregion

        #region PROPERTIES
        public int Floor 
        {
            get { return _floor; }
            private set
            {
                if (value > Battery.NumFloors || value < -(Battery.NumBasements))
                    throw new Exception("The floor value provided for the request is invalid.");
                else
                    _floor = value;
            }
        }
        public string Direction
        {
            get { return _direction; }
            private set
            {
                if (value.ToLower() != "up" || value.ToLower() != "down")
                    throw new Exception("The direction provided for the request is invalid. It can only be 'up' or 'down'.");
                else
                    _direction = value;
            }
        }
        #endregion

        #region CONSTRUCTOR
        public Request(int floor_, string direction_)
        {
            Floor = floor_;
            Direction = direction_;
        }
        #endregion
    }
}

File: Commercial/Commercial_ControllerJava/src/Request.java
public class Request {

    //region FIELDS
    private int floor;
    private String direction;
    //endregion

    //region PROPERTIES - Getters
    public int getFloor() {
        return floor;
    }
    public String getDirection() {
        return direction;
    }
    //endregion

    //region PROPERTIES - Setters
    public void setFloor(int value)
    {
        if (value > Battery.NumFloors || value < -(Battery.NumBasements))
            throw new RuntimeException("The floor value provided for the request is invalid.");
        else
            floor = value;
    }
    public void setDirection(String value)
    {
        if (value.toLowerCase() != "up" && value.toLowerCase() != "down")
            throw new RuntimeException("The direction provided for the request is invalid. It can only be 'up' or 'down'.");
        else
            direction = value;
    }
    //endregion

    //region CONSTRUCTOR
    public Request(int floor, String direction)
    {
        setFloor(floor);
        setDirection(direction);
    }
    //endregion

}
